src/model/SimulationControl.cpp

```diff
@@ -169,29 +179,94 @@ namespace model {
     }
 
     bool SimulationControl_Impl::doZoneSizingCalculation() const {
-      boost::optional<std::string> value = getString(OS_SimulationControlFields::DoZoneSizingCalculation, true);
-      OS_ASSERT(value);
-      return openstudio::istringEqual(value.get(), "Yes");
+
+      auto shouldItBeOn = [this]() -> bool {
+        const auto m = model();
+        const auto hasSizingPeriods = !m.getModelObjects<SizingPeriod>().empty();
+        if (!hasSizingPeriods) {
+          return false;
+        }
+
+        const auto zones = m.getConcreteModelObjects<ThermalZone>();
+        const auto hasZoneEquipment = std::any_of(zones.cbegin(), zones.cend(), [](const auto& z) { return !z.equipment().empty(); });
+
+        return hasZoneEquipment;
+      };
+
+      if (!isDoZoneSizingCalculationDefaulted()) {
+        boost::optional<std::string> value = getString(OS_SimulationControlFields::DoZoneSizingCalculation, true);
+        OS_ASSERT(value);
+        const auto result = openstudio::istringEqual(value.get(), "Yes");
+        if (!result && shouldItBeOn()) {
+          LOG(Warn, "You have zonal equipment and design days, it's possible you should enable SimulationControl::DoZoneSizingCalculation");
+        }
+        return result;
+      }
+
+      const auto result = shouldItBeOn();
+      if (result) {
+        LOG(Debug, "You have zonal equipment and design days, and SimulationControl::DoZoneSizingCalculation is defaulted: turning on.");
+      }
+
+      return result;
```

Comment:
Example with doZoneSizing: the model object has smart logic.

* If not defaulted, but we think it should be on: warn
* If defaulted, we check whether we think it should be on. If so, since this is the opposite of the IDD default, we issue a Debug message explaining why we picked it